datadog/tests/data_source_datadog_user_roles_test.go

```diff
@@ -0,0 +1,114 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccDatadogUserRolesDatasourceBasic(t *testing.T) {
+	t.Parallel()
+	ctx, _, accProviders := testAccFrameworkMuxProviders(context.Background(), t)
+	uniq := strings.ToLower(uniqueEntityName(ctx, t))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: accProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceUserRolesConfig(uniq),
+				Check:  resource.TestCheckResourceAttrSet("data.datadog_user_roles.foo", "user_roles.0.role_id"),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUserRolesDatasourceExactMatch(t *testing.T) {
+	t.Parallel()
+	ctx, _, accProviders := testAccFrameworkMuxProviders(context.Background(), t)
+	uniq := strings.ToLower(uniqueEntityName(ctx, t))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: accProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceUserRolesExactMatchConfig(uniq, "false"),
+				Check:  resource.TestCheckResourceAttr("data.datadog_user_roles.ur", "user_roles.#", "2"),
+			},
+			{
+				Config: testAccDatasourceUserRolesExactMatchConfig(uniq, "true"),
+				Check:  resource.TestCheckResourceAttr("data.datadog_user_roles.ur", "user_roles.#", "1"),
+			},
+		},
+	})
+}
+
+func testAccDatasourceUserRolesConfig(uniq string) string {
+	return fmt.Sprintf(`
+resource "datadog_user" "foo" {
+	email = "%s@example.com"
+	lifecycle {
+		ignore_changes = [ roles ]
+	}
```

Comment:
```suggestion
```

Lets instead mark the `roles` field in datadaog_user as `Optional+Computed`.  And add an additional description stating the roles field and datadog_user_role should not be used together